### PR TITLE
Vue new addressbook

### LINKED ISSUE #605: [Vue] Add addressbook
- [x] Store
- [x] Design 

---
- [ ] Security checks
- [ ] Regex

### PR DESCRIPTION
**Pull request by suntala**
- Add ability to create a new address book. 
- Fix style of submit button. 
- Include an initial definition for addressBookRegex data property. 

Fix #605. Regex/security checks are still required. In particular there is still an issue with checking for whitespace.

### CODE REVIEW

File: src/components/Settings/SettingsNewAddressbook.vue [hunk 1]
```diff
@@ -68,14 +62,12 @@ export default {
...
+		 *
 		 */
-		addAddressBook() {
-			let addressBook = this.$refs.addressBook.value
-			let addressBooks = this.$store.getters.getAddressbooks
-			let newAddressBooksArray = addressBooks.push(addressBook)
-			return newAddressBooksArray
+		addAddressbook() {
+			let addressbook = this.$refs.addressbook.value
+			this.$store.dispatch('appendAddressbook', { displayName: addressbook })
```
Comment (skjnldsv) [inline src/components/Settings/SettingsNewAddressbook.vue]: :+1: Nice with the object that gets extended in the store!

File: src/components/Settings/SettingsNewAddressbook.vue [hunk 2]
```diff
@@ -47,16 +48,9 @@ export default {
...
-	// 	addressbooks: {
-	// 		type: Array,
-	// 		default() {
-	// 			return {}
-	// 		}
-	// 	}
-	// },
 	data() {
 		return {
+			addressBookRegex: new RegExp('/^[a-zA-Z0-9À-ÿ\s_.!?#|()-]+$/i')
```
Comment (skjnldsv) [inline src/components/Settings/SettingsNewAddressbook.vue]: I'd say, let's drop this for now. We'll see later on :)
(leave it empty we'll fix it later)

File: src/components/Settings/SettingsNewAddressbook.vue (hunk 1 above)
Comment (suntala) [reply src/components/Settings/SettingsNewAddressbook.vue]: Thanks! :)

File: src/components/Settings/SettingsNewAddressbook.vue [hunk 3]
```diff
@@ -68,14 +62,12 @@ export default {
 		/**
 		 * Set new address book name
 		 *
-		 * @param {string} addressBook The adress book
-		 * @returns {Promise}
+		 * @param {string} addressbook The address book name provided in the input
```
[the remark this answers is not in the document]
Comment (suntala) [reply src/components/Settings/SettingsNewAddressbook.vue]: Yes, good point.

File: src/components/Settings/SettingsNewAddressbook.vue [hunk 4]
```diff
@@ -21,18 +21,19 @@
...
 
 <template>
-	<form id="new-addressbook-form" name="new-addressbook-form" class="new-address-book"
-		@submit.prevent="addAddressBook">
-		<input id="new-address-book" ref="addressBook" class="new-address-book-input"
+	<form id="new-addressbook-form" name="new-addressbook-form" class="new-addressbook"
+		@submit.prevent.stop="addAddressbook">
+		<input id="new-addressbook" ref="addressbook" :pattern="addressBookRegex"
+			class="new-addressbook-input"
 			placeholder="Address book name"
```
[the remark this answers is not in the document]
Comment (suntala) [reply src/components/Settings/SettingsNewAddressbook.vue]: Oh yes, that's right!

File: src/components/Settings/SettingsNewAddressbook.vue [hunk 5]
```diff
@@ -21,18 +21,15 @@
...
 			type="text"
 			autocomplete="off" autocorrect="off"
-			spellcheck="false"
-			tooltip-enable="!newAddressBookForm.$pristine"
-			tooltip-trigger="none"
-			tooltip-placement="top"
-			uib-tooltip="Only these special characters are allowed: -_.!?#|()">
-		<input type="submit" value="" class="newAddressBookSubmit inline-button icon-confirm action pull-right">
+			spellcheck="false">
+		<input type="submit" value="" class="newAddressbookSubmit inline-button icon-confirm action pull-right">
```
Comment (skjnldsv) [inline src/components/Settings/SettingsNewAddressbook.vue]: newAddressbookSubmit should probably be written with bem syntax?
Also: `inline-button`, `action` and `pull-right` are not needed :)

File: src/components/Settings/SettingsNewAddressbook.vue [hunk 6]
```diff
@@ -47,16 +44,10 @@ export default {
...
-	// 		type: Array,
-	// 		default() {
-	// 			return {}
-	// 		}
-	// 	}
-	// },
 	data() {
 		return {
+			// match any expression
+			addressBookRegex: new RegExp('/.*/')
```
Comment (skjnldsv) [inline src/components/Settings/SettingsNewAddressbook.vue]: :+1:

Comment (suntala) [reply src/components/Settings/SettingsNewAddressbook.vue]: Hmm this does not seem to be working anymore 😞. 
It works on here: https://regex101.com/r/cO8lqs/2
I need to look at it some more.

Comment (skjnldsv) [reply src/components/Settings/SettingsNewAddressbook.vue]: Remove the regex from the input then and add a `TODO:` stating we need to add a check later :)

**Pull request merged**